File: Lab5/PythonLab5/TPyramid.py
from TFigure import TFigure
from math import pow
from math import sqrt
import random
#правильна піраміда (в основі лежить правильний багатокутник)
class TPyramid(TFigure):
    def __init__(self, Basis, Side, Height):
        self.basis = Basis      #форма основи піраміди
        self.side = Side        #довжина сторони основи піраміди
        self.height = Height    #висота піраміди
    
    #конструктор за замовчуванням, випадкові значення
    def __init__(self):
        self.randomValue()

    #площа поверхні піраміди
    def area(self):
        if self.basis == "triangle":
            return pow(self.side, 2) * sqrt(3) / 4 + 1.5 * self.side * sqrt(pow((self.side / (2 * sqrt(3))), 2) + pow(self.height, 2))
        elif self.basis == "square":
            return self.side * self.side + 2 * self.side * sqrt(pow((self.side / 2), 2) + pow(self.height, 2))
    
    #об'єм піраміди
    def volume(self):
        if self.basis == "triangle":
            return pow(self.side, 2) * sqrt(3) / 4 * self.height / 3
        elif self.basis == "square":
            return self.side * self.side * self.height / 3
    
    #випадкове значення піраміди
    def randomValue(self):
        self.basis = random.choice(["triangle", "square"])
        self.side = random.randrange(1, 10)
        self.height = random.randrange(1, 10)
    
    #виведення інформації про піраміду
    def output(self):
        print("Pyramid: {} h, {} s, {}".format(self.height, self.side, self.basis))

File: Lab5/CPPLab5/TPyramid.cpp
#include "TPyramid.h"

//перелічувальний тип для позначення основи піраміди
enum class figure
{
	triangle = 3,	//основа - рівносторонній трикутник
	square = 4		//основа - квадрат
};

//перевірка форми основи піраміди
string figureCheck(figure f)
{
	return f == figure::triangle ? "triangle" : "square";
}

//площа поверхні піраміди
float TPyramid::area()
{
	switch (basis)
	{
	case figure::triangle:
		return pow(side, 2) * sqrt(3) / 4 + 1.5 * side * sqrt(pow((side / (2 * sqrt(3))), 2) + pow(height, 2));
	case figure::square:
		return side * side + side * sqrt(pow((side / 2), 2) * 2 + pow(height, 2));
	}
}

//об'єм піраміди
float TPyramid::volume()
{
	switch (basis)
	{
	case figure::triangle:
		return pow(side, 2) * sqrt(3) / 4 * height / 3;
	case figure::square:
		return side * side * height / 3;
	}
}

//випадкове значення піраміди
void TPyramid::randomValue()
{
	basis = figure(rand() % 2 + 3);
	side = rand() % 10 + 1;
	height = rand() % 10 + 1;
}

//виведення інформації про піраміду
void TPyramid::output()
{
	cout << "\nPyramid: " << setw(2) << height << " h, " << setw(2) << side << " s, " << figureCheck(basis);
}